pkgs/servers/http/nginx/generic.nix

```diff
@@ -158,9 +158,16 @@ stdenv.mkDerivation {
     cp -r ${nginx-doc}/* $doc
   '';
 
-  postInstall = if postInstall != null then postInstall else ''
-    mv $out/sbin $out/bin
-  '';
+  nativeBuildInputs = [ removeReferencesTo ];
+
+  disallowedReferences = map (m: m.src) modules;
+
+  postInstall =
+    let
+      noSourceRefs = lib.concatMapStrings (m: "remove-references-to -t ${m.src} $out/sbin/nginx\n") modules;
+    in noSourceRefs + (if postInstall != null then postInstall else ''
+      mv $out/sbin $out/bin
```

Comment:
why is this needed?

Reply:
What exactly do you mean? The changed part implements exactly what's described in this PR and the commit message.

Reply:
>       mv $out/sbin $out/bin

Reply:
Ah, ok, that line. It has only been moved, so it's really not my idea. Don't all derivations provide their executables in 'bin' only? Not sure, sorry.

Reply:
Ah, don't worry. Didn't read the diff well enough.

> Don't all derivations provide their executables in 'bin' only?

They do, but: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/move-sbin.sh

This whole thing is kind of weird, tbh. If you feel like it's out of scope, we can go ahead as is, but we could also just change the `postInstall ? null` to `postInstall ? ""` and remove this `mv` and replace the whole `if … then … else` thing with a simple `+`. Although… maybe that needs a `\n` before the `+`? Is that why you're using `concatMapStrings` instead of `concatMapStringsSep`?

Reply:
Thanks for the suggestion, I cleaned up the `postInstall` phase accoringly in another commit.